src/Integration/LocalServices/ProjectSystemFilter.cs

```diff
@@ -85,13 +72,58 @@ public bool IsAccepted(DteProject dteProject)
 
         public void SetTestRegex(Regex regex)
         {
+            if (regex == null)
+            {
+                throw new ArgumentNullException(nameof(regex));
+            }
+
             this.testRegex = regex;
-            Debug.Assert(this.testRegex == null || (this.testRegex.MatchTimeout != Regex.InfiniteMatchTimeout), "Should have set non-infinite timeout");
+            Debug.Assert(this.testRegex.MatchTimeout != Regex.InfiniteMatchTimeout, "Should have set non-infinite timeout");
         }
 
         #endregion
 
         #region Helpers
+        private static bool IsTestProject(IVsHierarchy projectHierarchy, Regex testProjectNameRegex, string projectName)
```

Comment:
The method name suggests that all of the test-detection code is in this method, but it isn't. The processing would be easier to follow if the _SonarQubeTestProject_ code was here too.